File: github_code_index_sort/cpp/8/60.c
#include <stdio.h>

int Partition(int *A,int p,int r)
{
    int x=A[r];
    int i=p-1;
    int temp=0;
    for(int j=p;j<=(r-1);j++)
    {
        if(A[j]<=x)
        {
            i+=1;
            temp=A[i];
            A[i]=A[j];
            A[j]=temp;
        }
        
        
    }
     temp=A[i+1];
     A[i+1]=A[r];
     A[r]=temp;
     
    return i+1;
    
}

void Quicksort(int *A,int p,int r)
{
    int q=0;
    if(p<r)
    {
        q=Partition(A,p,r);
    
    
    Quicksort(A,p,q-1);
    Quicksort(A,q+1,r);
    }
}

int main()
{
   int A[9]={7,5,3,4,6,8,7,1,5};
   Quicksort(A,0,8);
   for(int i=0;i<9;i++)
   {
        printf("    %d\n", A[i]);
   }
    return 0;
}

File: github_java_sort/10/60.java
package Array;


public class FirstMissingPositive {
	public int firstMissingPositive(int[] A) {
		int n = A.length;
		for (int i = 0; i < n; i++) {
			
			while (A[i] != i) {
				
				if (A[i] < 0 || A[i] >= n)
					break;
	 
				
				if(A[i]==A[A[i]])
	                    		break;
				
				int temp = A[i];
				A[i] = A[temp];
				A[temp] = temp;
			}
		}
	 
		for (int i = 0; i < n; i++) {
			if (A[i] != i)
				return i;
		}
	 
		return n;
	}
}
